src/main/java/com/future/function/qa/data/question_bank/QuestionBankData.java

```diff
@@ -0,0 +1,57 @@
+package com.future.function.qa.data.question_bank;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.future.function.qa.data.BaseData;
+import com.future.function.qa.model.request.question_bank.QuestionBankWebRequest;
+import com.future.function.qa.model.response.base.DataResponse;
+import com.future.function.qa.model.response.base.PagingResponse;
+import com.future.function.qa.model.response.question_bank.QuestionBankWebResponse;
+import io.restassured.response.Response;
+import java.util.Objects;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Data
+@Builder
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+public class QuestionBankData extends BaseData {
+
+  private PagingResponse<QuestionBankWebResponse> pagedResponse = new PagingResponse<>();
+
+  private QuestionBankWebRequest request;
+
+  private DataResponse<QuestionBankWebResponse> singleResponse = new DataResponse<>();
+
+  public QuestionBankWebRequest createRequest(String id, String title, String description) {
+
+    request = QuestionBankWebRequest.builder()
+        .title(title)
+        .description(description)
+        .build();
+
+    if (id != null) {
+      request.setId(id);
+    }
+
+    return request;
+  }
+
+  @Override
+  public void setResponse(Response response) {
+    super.setResponse(response);
+    this.pagedResponse = Optional.of(response)
```

Comment:
Just asking, why use `Optional` here?

Reply:
prevent NPE because when accessing the data, it can be null. As you can see, there is a filter `Objects.nonNull(res.getData)` and if the data is null, then it will return empty pagedResponse